### PR TITLE
modifying to read from env then fallback to database.json

### PR DESCRIPTION
**Pull request by ddombrow**
Summary
- Made database.json reading separated by environment keys
- Made the config prefer the .env settings
- Changed the name of the .env settings to avoid conflict
- Now uses dotenv to load .env settings from file if not there

### CODE REVIEW

**Comment by ddombrow** [comment]
@danmactough what you had in mind?

File: lib/migrate.js [hunk 1]
```diff
@@ -69,20 +70,26 @@ function wait(connection, options) {
...
-    authKey: nconf.get('authKey'),
-    ssl: nconf.get('ssl')
+  var config;
+  config = {
+    host: nconf.get("RETHINK:HOST"),
+    port: nconf.get("RETHINK:PORT") || 28015,
+    db: nconf.get("RETHINK:DB"),
+    timeout: nconf.get("RETHINK:TIMEOUT") || 60,
+    authKey: nconf.get("RETHINK:AUTHKEY")
   };
```
Comment (danmactough) [inline lib/migrate.js]: What about discovery and ssl?

We've also changed the default timeout from 5s to 1s. Intentional? Could that cause problems?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rethink-migrate",
-  "version": "1.3.0",
+  "version": "2.0.1",
```
Comment (danmactough) [inline package.json]: Not `2.0.0`?

File: lib/migrate.js (hunk 1 above)
Comment (ddombrow) [inline lib/migrate.js]: discovery and ssl were never used.  I don't even know what they would mean.  The default timeout is getting changed back, not intentional.

File: package.json (hunk 2 above)
Comment (ddombrow) [inline package.json]: just npm versioning as I go, there was a patch after 2.0.0

File: lib/migrate.js (hunk 1 above)
Comment (danmactough) [inline lib/migrate.js]: gotcha. thanks.

File: package.json (hunk 2 above)
Comment (danmactough) [inline package.json]: hmm. usually we don't version or tag on branches at all. only on master.

**Comment by danmactough** [comment]
I like this.

**Comment by ddombrow** [comment]
@danmactough so... ready to merge?

**Comment by danmactough** [comment]
@ddombrow Can we undo the version bumping in the branch so we can stay consistent?

**Pull request merged**